internal/cmdrender/executor.go

```diff
@@ -540,3 +560,105 @@ func pruneResources(hctx *hydrationContext) error {
 	}
 	return nil
 }
+
+// selectInput returns the selected resources based on criteria in selectors
+func selectInput(input []*yaml.RNode, selectors []kptfilev1.Selector) []*yaml.RNode {
+	if len(selectors) == 0 {
+		return input
+	}
+	var filteredInput []*yaml.RNode
+	for _, selector := range selectors {
+		for _, node := range input {
+			if (selector.Name == "" || selector.Name == node.GetName()) &&
+				(selector.Namespace == "" || selector.Namespace == node.GetNamespace()) &&
+				(selector.APIVersion == "" || selector.APIVersion == node.GetApiVersion()) &&
+				(selector.Kind == "" || selector.Kind == node.GetKind()) {
+				filteredInput = append(filteredInput, node)
+			}
+		}
+	}
+	return filteredInput
+}
+
+// setResourceIds adds resource-id annotation to each input resource
```

Comment:
We should include why we're using resource IDs for tracking either in these comments or above.

Reply:
+1

Reply:
Added comment above resource-id annotation.